### PR TITLE
feat(generic) support generic map

### PR DESCRIPTION
**Pull request by gxcsoccer**
- support encode es6 Map to java generic map

issue #44

### CODE REVIEW

**Comment by fengmk2** [comment]
Test fail on 0.10

File: test/map.test.js [hunk 1]
```diff
@@ -161,6 +161,23 @@ describe('map.test.js', function () {
     });
   });
 
+  it('should write es6 Map to java.util.Map', function() {
+    var map = new Map();
```
Comment (dead-horse) [inline test/map.test.js]: 0.10 测试会跪，要么把这个单独拎出来，要么0.10不测了吧

Comment (gxcsoccer) [inline test/map.test.js]: 我加个判断吧

**Comment by fengmk2** [comment]
+1

**Comment by fengmk2** [comment]
hessian.js@2.1.0

**Pull request merged**